maestro-studio/server/src/main/java/maestro/studio/DeviceScreenService.kt

```diff
@@ -78,20 +81,23 @@ object DeviceScreenService {
         val elements = gatherElements(tree, mutableListOf())
             .sortedWith(Filters.INDEX_COMPARATOR)
 
-        fun getIndex(element: TreeNode, attribute: String): Int? {
-            val value = element.attribute(attribute) ?: return null
-            val matchingElements = elements.filter { value == it.attributes[attribute] }
+        fun getIndex(filter: ElementFilter, element: TreeNode): Int? {
+            val identityHashMap = IdentityHashMap<TreeNode, Unit>()
+            val matchingElements = Filters.deepestMatchingElement(filter)(elements).filter {
+                // There are duplicate elements for some reason (likely due to unintended behavior in Filter.deepestMatchingElement) - filter them out
```

Comment:
Duplicate elements are sometimes coming with the hierarchy, especially on iOS

Reply:
@dmitry-zaitsev The _objects_ are actually being duplicated by the filter so this is a bug in our code